### DISCUSSION TITLE
factory method on relations

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#40814] [Ideas] [answered]
```php
$customer = Customer::factory()->create();

$customer->addresses()->factory()->create(['city' => 'Miami']);
$customer->addresses()->factory()->create(['city' => 'New York']);
```

We know the Address model, its factory, and the value of the `customer_id`

Would be cool if Laravel added a factory method on relationship classes to set this up

**Comment by chuckrincon** [comment, marked as answer]
Hi there ✋🏻 

> Would be cool if Laravel added a factory method on relationship classes to set this up

Actually we have this feature, you could try using  the [Factory Relationships](https://laravel.com/docs/8.x/database-testing#factory-relationships) and the [Sequences](https://laravel.com/docs/8.x/database-testing#sequences) to get the same result:

```php
Customer::factory()
    ->has(
        Address::factory(2)
            ->state(new Sequence(
                ['city' => 'Miami'],
                ['city' => 'New York'],
            ))
    )
    ->create();
```

**Reply by dillingham** [reply]
Hey @chuckrincon 👋

That is really cool, I should spend more time with sequences.. never used them.

I appreciate you taking the time to show me that.. very slick solution
